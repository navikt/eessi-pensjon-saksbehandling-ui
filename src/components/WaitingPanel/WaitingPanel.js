import React from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import { NavFrontendSpinner } from 'components/Nav'

const WaitingPanel = (props) => {
  const { message, className } = props

  return <div className={classNames('c-waitingPanel', 'text-center', className)}>
    <NavFrontendSpinner />
    <p className='c-waitingPanel__message typo-normal'>{message}</p>
  </div>
}

WaitingPanel.propTypes = {
  message: PT.string.isRequired,
  className: PT.string
}

export default WaitingPanel
