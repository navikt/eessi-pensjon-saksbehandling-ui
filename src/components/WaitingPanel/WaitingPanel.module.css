.rowDirection {
  flex-direction: row !important;
}
