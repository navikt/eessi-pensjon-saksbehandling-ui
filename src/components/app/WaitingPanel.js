import React, { Component } from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import { withNamespaces } from 'react-i18next'
import * as Nav from '../ui/Nav'

class WaitingPanel extends Component {
  render () {
    const { t, message } = this.props

    return <div className={classNames('text-center')}>
      <Nav.NavFrontendSpinner />
      <p className='typo-normal'>{t(message)}</p>
    </div>
  }
}

WaitingPanel.propTypes = {
  message: PT.string.isRequired
}

export default withNamespaces(WaitingPanel)
