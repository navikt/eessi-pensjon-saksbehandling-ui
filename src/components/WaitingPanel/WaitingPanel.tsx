import { HorizontalSeparatorDiv } from 'components/StyledComponents'
import React from 'react'
import PT from 'prop-types'
import classNames from 'classnames'
import Spinner from 'nav-frontend-spinner'
import { Normaltekst } from 'nav-frontend-typografi'
import './WaitingPanel.css'
import styled from 'styled-components'

export interface WaitingPanelProps {
  className?: string;
  size?: 'XXS'| 'XS' | 'S'| 'M' | 'L'| 'XL'| 'XXL' | 'XXXL';
  style?: React.CSSProperties;
  message?: string,
  oneLine?: boolean
}

const WaitingPanelDiv = styled.div`
  text-align: center;
  .oneLine {
    display: inline-block;
    vertical-align: top;
  }
`

const WaitingPanel: React.FC<WaitingPanelProps> = ({
  className, size = 'M', style = {}, message = 'Vennligst vent...', oneLine = false
}: WaitingPanelProps): JSX.Element | null => (
  <WaitingPanelDiv
    style={style}
    className={className}
  >
    <Spinner type={size} />
    {message && (
      <>
        <HorizontalSeparatorDiv/>
        <Normaltekst
          className={classNames({ oneLine: oneLine })}
        >
          {message}
        </Normaltekst>
      </>
    )}
  </WaitingPanelDiv>
)

WaitingPanel.propTypes = {
  className: PT.string,
  message: PT.string,
  oneLine: PT.bool,
  size: PT.oneOf(['XXS', 'XS', 'S', 'M', 'L', 'XL', 'XXL', 'XXXL']),
  style: PT.object
}
WaitingPanel.displayName = 'WaitingPanel'
export default WaitingPanel
