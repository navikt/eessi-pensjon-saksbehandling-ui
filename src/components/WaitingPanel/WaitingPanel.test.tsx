import { mount, ReactWrapper } from 'enzyme'
import WaitingPanel, { WaitingPanelDiv } from './WaitingPanel'

describe('components/WaitingPanel/WaitingPanel', () => {
  let wrapper: ReactWrapper

  it('Render: match snapshot', () => {
    wrapper = mount(<WaitingPanel message='' />)
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    wrapper = mount(<WaitingPanel message='testmessage' />)
    expect(wrapper.exists(WaitingPanelDiv)).toBeTruthy()
    expect(wrapper.find('[data-test-id=\'c-waitingpanel__text-id\']').hostNodes().text()).toEqual('testmessage')
  })
})
