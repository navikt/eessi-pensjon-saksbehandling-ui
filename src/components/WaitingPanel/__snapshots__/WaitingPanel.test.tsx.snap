// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/WaitingPanel/WaitingPanel Render: match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1.rowDirection {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 .oneLine {
  display: inline-block;
  vertical-align: top;
}

ui-fss.g40[id="c0"] {
  content: "c1,";
}

<div
  class="c0 c1"
  data-testid="c-WaitingPanel"
>
  <svg
    aria-labelledby="loader-:r0:"
    class="navds-loader navds-loader--medium navds-loader--neutral"
    focusable="false"
    preserveAspectRatio="xMidYMid"
    type="medium"
    viewBox="0 0 50 50"
  >
    <title
      id="loader-:r0:"
    >
      venter...
    </title>
    <circle
      class="navds-loader__background"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      xmlns="http://www.w3.org/2000/svg"
    />
    <circle
      class="navds-loader__foreground"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      stroke-dasharray="50 155"
    />
  </svg>
</div>
`;
